rotkehlchen/tests/unit/decoders/test_1inch.py

```diff
@@ -163,6 +168,65 @@ def test_1inchv2_swap_for_eth(database, ethereum_inquirer):
     assert expected_events == events
 
 
+@pytest.mark.parametrize('ethereum_accounts', [['0x312419eEC9C4632155904D9440dc1EeeafFBb280']])
+def test_1inchv4_swap_on_uniswapv3(database, ethereum_inquirer):
+    """
+    Test an 1inch v4 swap for ETH via Uniswap v3.
+
+    Data taken from
+    https://etherscan.io/tx/0xd02bbee01f92d778af8c2d159fb269ad31425b32703da568abb427ac14547e6d
+    """
+    tx_hash = deserialize_evm_tx_hash('0xd02bbee01f92d778af8c2d159fb269ad31425b32703da568abb427ac14547e6d')  # noqa: E501
+    events, _ = get_decoded_events_of_transaction(
+        evm_inquirer=ethereum_inquirer,
+        database=database,
+        tx_hash=tx_hash,
+    )
+    timestamp = TimestampMS(1687361999000)
+    user_address = '0x312419eEC9C4632155904D9440dc1EeeafFBb280'
+    expected_events = [
+        EvmEvent(
+            tx_hash=tx_hash,
+            sequence_index=0,
+            timestamp=timestamp,
+            location=Location.ETHEREUM,
+            event_type=HistoryEventType.SPEND,
+            event_subtype=HistoryEventSubType.FEE,
+            asset=A_ETH,
+            balance=Balance(amount=FVal('0.005753512813468296')),
+            location_label=user_address,
+            notes='Burned 0.005753512813468296 ETH for gas',
+            counterparty=CPT_GAS,
+        ), EvmEvent(
+            tx_hash=tx_hash,
+            sequence_index=1,
```

Comment:
The order is wrong here. Swaps **always** have the out event first and then then in event. Otherwise accounting and other assumptions we make in the way we analyze events break down.